Use CS10 meta content provider and tcib images for testing

raukadah opened the pull request:
This pr adds following things:

- Drops openstack-k8s-operators content provider,
- Modifies existing patch-openstack-versions to a template to override/use meta content provider tcib containers via
   telemetry_registry and telemetry_tag var.

 Signed-off-by: Chandan Kumar <raukadah@gmail.com>
 Co-Authored-By: Emma Foley <efoley@redhat.com>

Depends-On: https://github.com/openstack-k8s-operators/telemetry-operator/pull/715

raukadah commented:
Based on results from autoscaling job https://softwarefactory-project.io/zuul/t/rdoproject.org/build/e8e47f651e354714a06ab43887503dd3
aodh is using containers from content provider: https://logserver.rdoproject.org/e8e/rdoproject.org/e8e47f651e354714a06ab43887503dd3/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/aodh-0/aodh-0-describe
```
Containers:
  aodh-api:
    Container ID:  cri-o://c6a6dee59d83239e2b939c7e7ff8762e48b99121c432c92ad4e72c8a335e21aa
    Image:         38.102.83.51:5001/podified-master-centos10/openstack-aodh-api:watcher_latest
    Image ID:      38.102.83.51:5001/podified-master-centos10/openstack-aodh-api@sha256:f73aa4e389283df9bb7b3c9a805e7a14f7df491808cdfc2ba81911d87369db87
```

raukadah commented:
From meta content provider 
https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/
```
[ ]	[openstack-ceilometer-24.1.0-0.20250626083454.8d272cc.el10.src.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/openstack-ceilometer-24.1.0-0.20250626083454.8d272cc.el10.src.rpm)	2025-06-26 04:38 	887K
[ ]	[openstack-ceilometer-central-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/openstack-ceilometer-central-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)	2025-06-26 04:38 	8.1K
[ ]	[openstack-ceilometer-common-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/openstack-ceilometer-common-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)	2025-06-26 04:38 	41K
[ ]	[openstack-ceilometer-compute-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/openstack-ceilometer-compute-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)	2025-06-26 04:38 	8.1K
[ ]	[openstack-ceilometer-ipmi-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/openstack-ceilometer-ipmi-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)	2025-06-26 04:38 	9.3K
[ ]	[openstack-ceilometer-notification-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/openstack-ceilometer-notification-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)	2025-06-26 04:38 	13K
[ ]	[openstack-ceilometer-polling-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/openstack-ceilometer-polling-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)	2025-06-26 04:38 	8.7K
[ ]	[python3-ceilometer-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/python3-ceilometer-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)	2025-06-26 04:38 	298K
[ ]	[python3-ceilometer-tests-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm](https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/DLRN/data/repos/component/cloudops/current/python3-ceilometer-tests-24.1.0-0.20250626083454.8d272cc.el10.noarch.rpm)
```
and 
Same is installed in ceilometer container here https://logserver.rdoproject.org/ce1/rdoproject.org/ce11469d29324ce0a0dd79dea1f02c84/ci-framework-data/logs/8b2da301-a78e-47da-9653-e259e2f09597/base/os/ceilometer-base/ceilometer-base-build.log
```
 Package                        Arch    Version                                Repository                    Size
=================================================================================================================
Installing:
 openstack-ceilometer-common    noarch  1:24.1.0-0.20250626083454.8d272cc.el10 gating-repo                   41 k
```
DLRN built the ceilometer package from Depends-On https://review.opendev.org/c/openstack/ceilometer/+/910225 and the same is installed in ceilometer container.

elfiesmelfie commented:
@raukadah I think I've fixed the issue, and also rebased the change to get the latest master. Hopefully, it will work now.

raukadah commented:
> @raukadah I think I've fixed the issue, and also rebased the change to get the latest master. Hopefully, it will work now.

Thank you @elfiesmelfie for updating the pr, I just updated the pr to make it more readable. We now we have two vars to deal with registry. telemetry_registry and telemetry_tag.

In order to use cs10 current tag content, we can set telemetry_registry: quay.rdoproject.org/podified-master-centos10 and telemetry_tag to current in zuul job var, It will work.

elfiesmelfie commented on ci/files/containers.yaml:
```diff
@@ -0,0 +1,17 @@
+---
+# tcib container images used by cloudops team for telemetry testing
+container_images:
+- imagename: quay.io/podified-master-centos9/openstack-aodh-api:current-podified
```
Why are these using quay.io/podified-master-centos9 instead of quay.rdoproject.org/podified-master-centos10?

raukadah replied:
I have copied these names from tcib default container config https://github.com/openstack-k8s-operators/tcib/blob/main/container-images/containers.yaml, these urls are overridden by tcib tool while building the containers.

elfiesmelfie commented on ci/patch-openstack-versions.yaml.j2:
```diff
@@ -0,0 +1,14 @@
+---
+# telemetry_registry value should be quay.rdoproject.org/podified-centos10-master
```
```suggestion
# telemetry_registry value should be quay.rdoproject.org/podified-master-centos10
```

elfiesmelfie reviewed (commented):
We can move the patch-openstack-versions into telemetry-operator eventually, but for now, this looks good, and I'm willing to merge it.
